ui/src/lib/yjs/useYWorkflow.ts

```diff
@@ -147,29 +147,38 @@ export default ({
     (nodeIds: string[]) =>
       undoTrackerActionWrapper(() => {
         const workflowIds: string[] = [];
-
-        const removeNodes = (nodeIds: string[]) => {
-          nodeIds.forEach((nid) => {
-            if (nid === DEFAULT_ENTRY_GRAPH_ID) return;
-
-            const index = rawWorkflows.findIndex((w) => w.id === nid);
-            if (index === -1) return;
-
-            // Loop over workflow at current index and remove any subworkflow nodes
-            (rawWorkflows[index].nodes as Node[]).forEach((node) => {
+        nodeIds.forEach((nid) => {
+          if (nid === DEFAULT_ENTRY_GRAPH_ID) return;
+          workflowIds.push(nid);
+          const workflow = rawWorkflows.find((w) => w.id === nid);
+          const nodes = workflow?.nodes;
+          // Loop over workflow and remove any subworkflow nodes
+          if (nodes && Array.isArray(nodes)) {
```

Comment:
This change iterates on one level deep, and then stops. This is why removeNodes was using itself, to be able to reach ALL subworkflows, no matter how deep.
If you can console.log rawWorkflows and test to make sure when deleting subworkflows that contain subworkflows, that all appropriate workflows are removed on subworkflow node deletion 🙏🏻